### PR TITLE
Destructure code and add login.

### PR DESCRIPTION
**Pull request by yo9e5h**
- Destructure the code.
- Move secrets to .env
- Add bcrypt for password hash and verification.
- Login and register.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,133 @@
...
+
+# Stores VSCode versions used for testing VSCode extensions
+.vscode-test
+
+# yarn v2
+.yarn/cache
+.yarn/unplugged
+.yarn/build-state.yml
+.yarn/install-state.gz
+.pnp.*
```
Comment (IamSwap) [inline .gitignore]: Do we really need all of these paths in `.gitignore`?

File: routes/login.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    return res.status(400).json({ message: "User doesn't exists" });
+  }
+
+  const verified = await bcrypt.compareSync(user.password, req.body.password);
+
+  if (verified) {
+    return res.status(403).json({ message: "password did not match!" });
+  }
+
+  return res.json({ message: "Welcome aboard!" });
```
Comment (IamSwap) [inline routes/login.js]: Here we should return the authenticated user with an access token. 
That access token can be used for authenticating other endpoints.

File: utils/connection.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const { MongoClient } = require("mongodb");
+
+const dbClient = async () => {
+  const client = new MongoClient(process.env.DATABASE_URL, {
```
Comment (IamSwap) [inline utils/connection.js:4]: There should an `.env.example` file to set these variables.

**Review by IamSwap — CHANGES_REQUESTED** [review]

**Comment by yo9e5h** [comment]
- remove excess .gitignores
- add .env.example
- send token after register.

**Review by IamSwap — APPROVED** [review]

**Pull request merged**